libs/utils/src/properties.c

```diff
@@ -467,11 +468,42 @@ int celix_properties_size(const celix_properties_t *properties) {
 }
 
 celix_properties_iterator_t celix_propertiesIterator_construct(const celix_properties_t *properties) {
-    return hashMapIterator_construct((hash_map_t*)properties);
+    celix_properties_iterator_t iter;
+    hash_map_iterator_t mapIter = hashMapIterator_construct((hash_map_t*)properties);
+    iter._data1 = mapIter.map;
+    iter._data2 = mapIter.next;
+    iter._data3 = mapIter.current;
+    iter._data4 = mapIter.expectedModCount;
+    iter._data5 = mapIter.index;
```

Comment:
What about the following?

```C
CELIX_BUILD_ASSERT(sizeof(celix_properties_iterator_t) == sizeof(hash_map_iterator_t));
CELIX_BUILD_ASSERT(__alignof__(celix_properties_iterator_t) == __alignof__(hash_map_iterator_t));
```

So that we can safely cast `celix_properties_t*` to ` hash_map_iterator_t*` to avoid any runtime overhead.

Reply:
To be honest, I prefer the current way.

Maybe there is some overhead, but IMO the mapping is - a bit - more explicit.